.github/actions/generate-docs/action.yaml

```diff
@@ -0,0 +1,62 @@
+name: Generate Documentation
+description: Updates Sphinx-based documentation
+inputs:
+  sourceDirectory:
+    description: Path to repository checkout
+    required: false
+    default: ${{ github.workspace }}
+  disableLinkExtensions:
+    description: Disable Sphinx link extensions
+    required: false
+    default: 'false'
+runs:
+  using: composite
+  steps:
+    - name: Update Version Number and Copyright ↗️
+      id: setup
+      shell: bash
+      run: |
+        : Update Version Number and Copyright ↗️
+        if [[ "${RUNNER_DEBUG}" ]]; then set -x; fi
+
+        : "${major:=}"
+        : "${minor:=}"
+        : "${patch:=}"
+
+        read -r _ major _ minor _ patch _ <<< \
+          "$(grep -E -e "#define LIBOBS_API_(MAJOR|MINOR|PATCH)_VER *" libobs/obs-config.h \
+            | sed 's/#define //g' \
+            | tr -s ' ' \
+            | tr '\n' ' ')"
+
+        sed -i -E \
+          -e "s/version = '([0-9]+\.[0-9]+\.[0-9]+)'/version = '${major}.${minor}.${patch}'/g" \
+          -e "s/release = '([0-9]+\.[0-9]+\.[0-9]+)'/release = '${major}.${minor}.${patch}'/g" \
+          -e "s/copyright = '(2017-[0-9]+, Lain Bailey)'/copyright = '2017-$(date +"%Y"), Lain Bailey'/g" \
+          ${{ inputs.sourceDirectory }}/docs/sphinx/conf.py
+
+        if [[ '${{ inputs.disableLinkExtensions }}' == 'true' ]]; then
+          sed -i -e "s/html_link_suffix = None/html_link_suffix = ''/g" \
+           ${{ inputs.sourceDirectory }}/docs/sphinx/conf.py
+          echo "nameExtension=(No Extensions) " >> $GITHUB_OUTPUT
```

Comment:
That trailing space is by design, because it is composed in a compound statement in the upload artefact step:

`${{ steps.setup.outputs.nameExtension }}${{ steps.setup.outputs.commitHash }}`

This ensures that there is no double space if the output value is empty.

Reply:
Could you link me to the exact place(s) this comes up? This seems like an odd way to handle it.

Reply:
Line 59.

Reply:
Well, that's a bit unfortunate. I would almost suggest that nameExtension and commitHash are processed in a step into a title string or title string suffix, because I think having a trailing space here could cause confusion in the future (and someone will suggest removing it in the future not realizing that it's necessary). In my opinion, whitespace handling is the responsibility of a string builder, not of the individual components of a string. That said, it's not a critical issue, and I won't block on this here.

Reply:
I switched the code to generate the entire string placed before the commit hash in the script block, which should remedy this issue.

Reply:
Looks fine now. Thanks.